### ISSUE TITLE
[Regression] Users cannot enable E2E encryption on direct channels

### ISSUE THREAD

**Issue by Gummikavalier** [RocketChat/Rocket.Chat#20596]
### Description:

Users cannot enable E2E on direct channels anymore since 3.11.0. Still works on 3.9.7. Not tested on 3.10.x.

Enabling `Toggle Room E2E Encryption` privilege does not help either.
![Screenshot from 2021-02-04 09-40-32](https://user-images.githubusercontent.com/6958942/106860977-d0107380-66cd-11eb-90cb-76c0d25a5475.png)

Regardless of the permission, only users with admin role can see the option in direct rooms.

### Steps to reproduce:

1. Go to a direct channel
2. Click hamburger menu in the room settings
3. Look for E2E option

### Expected behavior:

You should see E2E option to enable it

### Actual behavior:

You cannot see E2E option
(sorry no screenshot, as the menu vanished always when I try to take one)

### Server Setup Information:

- Version of Rocket.Chat Server: 3.11.0
- Operating System: CentOS 7
- Deployment Method: tar
- Number of Running Instances: 20
- DB Replicaset Oplog: yes
- NodeJS Version: 12.18.4
- MongoDB Version: 3.6

### Client Setup Information

- Desktop App or Browser Version: Any browser

### Additional context

Previous similar issue that had been fixed before: https://github.com/RocketChat/Rocket.Chat/issues/17014

**Comment by tmartincpp** [comment]
Same issue for me: as an admin you can enable E2E in direct messages but not as a user or owner, despite the permissions being set.

**Comment by tmartincpp** [comment]
From my understanding the issue is that the code does not have an exception for direct messages, so it checks if the user have the right to edit rooms.

Here is a patch for _app/e2e/client/tabbar.ts_ (3.12.1) to make it work with direct messages : 
```
--- tabbar.ts.old	2021-03-30 17:09:55.473766370 +0200
+++ tabbar.ts	2021-03-30 17:11:26.000000000 +0200
@@ -11,7 +11,7 @@ addAction('e2e', ({ room }) => {
 	const e2eEnabled = useSetting('E2E_Enable');
 	const e2eReady = e2e.isReady() || room.encrypted;
 	const e2ePermission = room.t === 'd' || usePermission('toggle-room-e2e-encryption', room._id);
-	const hasPermission = usePermission('edit-room', room._id) && e2ePermission && e2eReady;
+	const hasPermission = ( usePermission('edit-room', room._id) || room.t === 'd' ) && e2ePermission && e2eReady;
 	const toggleE2E = useMethod('saveRoomSettings');
 
 	const action = useMutableCallback(() => 

{
```

Also for the current development code : 
https://github.com/RocketChat/Rocket.Chat/commit/42ff6fddf965209b72ad1db7a37aa3bd38f3fdd8

I'm actually not a developer, especially not a javascript one, so I hope this patch will looks OK to more experienced eyes.

I can confirm that it work with 3.12.1.

**Comment by ctr49** [comment]
It's working fine via API though. So it's really about the button not exposed, not the permissions missing:
curl -H @header.txt https://server.url/api/v1/rooms.saveRoomSettings -d '{ "rid": "room-id", "encrypted": true }'

**Comment by Gummikavalier** [comment]
Interestingly iOS-client can see the button for E2E, but does not let use to enable it. I was in the impression that mobile clients use API.

In addition to issue in enabling E2E by anyone else than admin status users, global setting
`E2E Encryption -> Enable encryption for Direct Rooms by default`
does not work either.
You can enable it but seeing from the database directly, the messages are not getting encrypted. I tested this with newly created direct channels after sending several messages between both online accounts with working keys (in private groups no issue).

**Comment by Gummikavalier** [comment]
Confirming fixed in RC 3.14.0. Closing the issue.
